File: recommendation_system/models/actor.py
from datetime import datetime

from models.person import Person


class Actor(Person):
    def __init__(self, id: int, first_name: str, last_name: str, date_of_birth: datetime, biography: str,
                 nationality: str, filmography: list, awards: list):
        super().__init__(id, first_name, last_name, date_of_birth, biography, nationality)
        self._filmography = filmography
        self._awards = awards

    def get_filmography(self):
        return self._filmography

    def set_filmography(self, filmography: list):
        self._filmography= filmography

    def get_awards(self):
        return self._awards

    def set_awards(self, awards: list):
        self._awards = awards

File: movie/src/main/java/com/example/movie/model/Actor.java
package com.example.movie.model;


import jakarta.persistence.Entity;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToMany;

import java.util.Date;
import java.util.List;
import java.util.UUID;

@Entity
public class Actor extends Person {

    @ManyToMany
    protected List<Movie> filmography;
    @OneToMany
    protected List<Award> awards;

    public Actor(String firstName, String lastName, Date dateOfBirth, String nationality, String biography, UUID profileImage, List<Movie> filmography, List<Award> awards) {
        super(firstName, lastName, dateOfBirth, nationality, biography, profileImage);
        this.filmography = filmography;
        this.awards = awards;
    }

    public Actor() {

    }

    public List<Movie> getFilmography() {
        return filmography;
    }

    public void setFilmography(List<Movie> filmography) {
        this.filmography = filmography;
    }

    public List<Award> getAwards() {
        return awards;
    }

    public void setAwards(List<Award> awards) {
        this.awards = awards;
    }

    @Override
    public String toString() {
        return "Actor{" +
                "filmography=" + filmography +
                ", awards=" + awards +
                '}';
    }
}
